### PR TITLE
feat(sketch): add support for generating icon ranges

### LINKED ISSUE #260: [Icon kit] Icons not rendering properly from plugin
- Alphanumeric numbers need to be filled not outlined in all sizes

### LINKED ISSUE #5694: Update icon generation command to use source assets
We currently use our optimized assets when generating icons in Sketch and this is causing several rendering issues, such as #4053 and #5568. This also causes some issues on website where people are downloading icons and putting them into Sketch.

As a result, we'd like to move towards using our source assets to generate Sketch symbols. This work will include:

- Authoring an extension to our metadata package to support adding asset source information to metadata
- Including this extension in generated assets for the plugin
- Update the plugin command to use the source asset instead of taking the build info and generating the source from that descriptor

This work will also include:

- Coordinating with website around the addition of this extension and working with them to see how we can expose this in the download as opposed to the current solution

### LINKED ISSUE #5695: Output 16/20 and 24/32 libraries for icons
We current output a single library for all icon symbols. Ideally we would split out our libraries into 16/20 and 24/32 for our icons.

### PR DESCRIPTION
**Pull request by joshblack**
Closes #5694
Closes #5695
Closes https://github.com/carbon-design-system/carbon-design-kit/issues/260

Addresses deprecation bit of: #4357 (we no longer generate deprecated icons)

This patch adds support for rendering both `[32, 24]` and `[20, 16]` sets of icons, alongside addressing previous rendering bugs by adding an extension to `icon-build-helpers` that embeds the source of each asset into metadata. As a result, we now use the source of the icon to render the symbol in Sketch.

#### Changelog

**New**

- `icon-build-helpers`
  - New `assets` extension exposed to library consumers to embed source and path information for the assets of an icon
- `sketch`
  - Add two new commands for rendering `[32, 24]` and `[20, 16]` icon sets

**Changed**

- `icon-build-helpers`
  - Update arguments of `build` to take in `input` and `output` paths to specify a custom input/output directory
- `sketch`
  - Generate local metadata in the `generated` folder with extensions for `pictograms` and `icons`
  - Use this metadata instead of `build-info.json` in our sync commands for icon symbols
  - Use the asset source for icon symbols instead of the optimized source from SVGO
- `icons`
  - Fix SVG assets that had extra paths that were causing incorrect rendering behavior when we merge paths

**Removed**

- `sketch`
  - Remove existing icon symbol sync command

### CODE REVIEW

**Comment by joshblack** [comment]
@abbeyhrt @dakahn let me know if you want to go over this!

**Comment by joshblack** [comment]
FYI @vpicone this is that assets extension work I was talking about. Nothing is changing for `metadata.json` from `@carbon/icons` though 👍 

The big thing most likely will be mirroring the rendering bugs when copying over into Sketch. Most likely we'll go back to optimizing the SVG assets now that we no longer use them in Sketch which might impact the library download capabilities.

File: packages/sketch/src/commands/icons/sync.js [hunk 1]
```diff
@@ -10,16 +10,32 @@ import { command } from '../command';
 import { syncIconSymbols } from './shared';
 import { findOrCreateSymbolPage } from '../../tools/page';
 
-export function sync() {
-  command('commands/icons/sync', () => {
+export async function syncSmallIcons() {
```
Comment (joshblack) [inline packages/sketch/src/commands/icons/sync.js]:
```suggestion
export function syncSmallIcons() {
```

File: packages/sketch/src/commands/icons/sync.js [hunk 2]
```diff
@@ -10,16 +10,32 @@ import { command } from '../command';
...
       Array.from(symbols),
       symbolsPage,
-      document.sharedLayerStyles
+      document.sharedLayerStyles,
+      [16, 20]
+    );
+  });
+}
+
+export async function syncLargeIcons() {
```
Comment (joshblack) [inline packages/sketch/src/commands/icons/sync.js]:
```suggestion
export function syncLargeIcons() {
```

**Comment by vpicone** [comment]
@joshblack nice, yeah thankfully the changes to the svgo config fixed our illustrator issues. If someone wants to use one of the SVG icons in sketch, then using the sketch addon would be the preferred workflow compares to downloading from the website anyhow.

**Review by emyarod — COMMENTED** [review]
should we ignore deprecated icons when we generate the icon page? currently the error for missing symbol is being thrown

**Comment by joshblack** [comment]
@emyarod ah yeah, definitely lol

**Review by laurenmrice — CHANGES_REQUESTED** [review]
- We don't need a transparent-rectangle or rectangle background on each icon artboard. The size of artboard the symbol sits on will make this container.
Screen Shot 2020-03-26 at 12 46 27 PM

-------------------------------------------------------------------------------------------------

- The `undo` and `redo` icons are rendering in a skewed way.
Screen Shot 2020-03-26 at 12 50 10 PM
Screen Shot 2020-03-26 at 1 02 25 PM

**Comment by joshblack** [comment]
@laurenmrice the only reason the rectangles show up is because they're in the original SVG assets 👍 can definitely remove them though.

Good catch on the arrows, let me figure out what's going on!

**Comment by laurenmrice** [comment]
let me know once you have another build. Could you also send me the 16,20px file so I can make sure they are consistent with the 24,32px

**Comment by joshblack** [comment]
@emyarod @laurenmrice updated!

**Review by abbeyhrt — APPROVED** [review]

**Review by dakahn — APPROVED** [review]
Cool work 🏄

**Comment by joshblack** [comment]
@laurenmrice confirmed that there were some issues still but that the issues are with the underlying SVG assets (not having transparent rectangle) so will move forward with addressing those in a separate issue!

**Review by laurenmrice — APPROVED** [review]

**Pull request merged**